batch-processing/README.md

```diff
@@ -0,0 +1,379 @@
+
+You can make it publically accessible to the internet, so that you or anyone can view the images that are uploaded, with a policy like this:
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "PublicAccess",
+      "Effect": "Allow",
+      "Principal": "*",
+      "Action": ["s3:GetObject", "s3:GetObjectVersion"],
+      "Resource": "arn:aws:s3:::[NAME OF BUCKET]/*"
+    }
+  ]
+}
+```
+
+> ⚠️ Public means public. **DO NOT** upload anything sensitive to the bucket.
+
+Use the [AWS SDK for Go](https://github.com/aws/aws-sdk-go) to interact with Amazon S3.
+
+You will need to set up credentials so that your docker image can write to S3. The best way to do that is to:
+
+- Create a role + policy to allow uploads to the bucket
+- Allow your AWS user account to use the role
+- Make your AWS credentials available to the Docker image
+
+#### Role + policy
+
+[Create a Policy](https://us-east-1.console.aws.amazon.com/iamv2/home#/policies) `S3ReadWriteGoCourse` with the following configuration, which allows reading and writing to the bucket.
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "ListObjectsInBucket",
+      "Effect": "Allow",
+      "Action": ["s3:ListBucket"],
+      "Resource": ["arn:aws:s3:::[NAME OF BUCKET]"]
+    },
+    {
+      "Sid": "AllObjectActions",
+      "Effect": "Allow",
+      "Action": "s3:*Object",
+      "Resource": ["arn:aws:s3:::[NAME OF BUCKET]/*"]
+    }
+  ]
+}
+```
+
+[Create a Role](https://us-east-1.console.aws.amazon.com/iamv2/home#/roles) `GoCourseLambdaUserReadWriteS3` associated with the `S3ReadWriteGoCourse` Policy. Set up the Trust Relationships of this policy as follows:
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "arn:aws:iam::[YOUR AWS USER ID]:root"
+      },
+      "Action": "sts:AssumeRole"
+    },
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+```
+
+One that's done, you need local credentials:
+
+1. Install the [AWS CLI](https://aws.amazon.com/cli/)
+1. [Follow this guide](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html) to set up your credentials locally
+
+Then change the `Makefile` to [mount](https://docs.docker.com/storage/bind-mounts/) your `.aws` directory into the Docker container, in the `root` user's home directory. When using the Go SDK, it will automatically find these credentials and use them:
+
+```Makefile
+run: outputs
+	docker ...
+        ...
+        --mount type=bind,source="$$(echo $$HOME)/.aws",target=/root/.aws \
+		...
+```
+
+You'll need to say which AWS region the S3 bucket is located in. Do that with the `AWS_REGION` environment variable.
+
+In your `Dockerfile`:
```

Comment:
Do you actually need to declare this in the `Dockerfile`? I think `-e` should work even if not?

Reply:
Perhaps not! In some iteration it had a value here.